cogs/basic.py

```diff
@@ -125,29 +125,38 @@ def __init__(self, bot):
 
     @commands.command()
     @commands.guild_only()
-    async def spam(self,ctx):
-        await ctx.send("You have requested spam.")
-        await asyncio.sleep(5)
-        await ctx.send("You have requested spam.")
-        await asyncio.sleep(5)
-        await ctx.send("You have requested spam.")
-        await asyncio.sleep(5)
-        await ctx.send("You have requested spam.")
-        await asyncio.sleep(5)
-        await ctx.send("You have requested spam.")
-        await asyncio.sleep(5)
-        await ctx.send("You have requested spam.")
-        await asyncio.sleep(5)
-        await ctx.send("You have requested spam.")
-        await asyncio.sleep(5)
-        await ctx.send("You have requested spam.")
-        await asyncio.sleep(5)
-        await ctx.send("You have requested spam.")
-        await asyncio.sleep(5)
-        await ctx.send("You have requested spam.")
+    @command.cooldown(1, 30, BucketType.user)
+    async def spam(self,ctx,amount:int=None):
```

Comment:
this entire thing is a bit of mess, why not just use an default value of 10? that makes the first if and else redundant and cleans this thing up quite a bit